File: Java/src/Search for a Range.java
public class Solution {
    public int[] searchRange(int[] A, int target) {
        int left = 0;
        int right = A.length - 1;
        int[] result = {-1, -1};
        while(left<right){
            int mid = (left + right) / 2;
            if(A[mid] < target) left = mid+1;
            else right = mid;
        }

        if (A[left] == target){
            result[0]= left;
            result[1]= left;
            while ((left+1 <= A.length-1) && A[left+1] == target) left++;
            result[1] = left;
        }
        return result;
    }
}

File: Python/Search for a Range.py
class Solution:
    # @param A, a list of integers
    # @param target, an integer to be searched
    # @return a list of length 2, [index1, index2]
    def searchRange(self, A, target):
        left = 0
        right = len(A) - 1
        ret = [-1, -1]
        while left < right:
            mid = (left + right)/2
            if A[mid]< target:
                left = mid + 1
            else:
                right = mid
        if A[left]==target:
            ret[0] = left
            while left <len(A)-1 and A[left+1]== target:
                left += 1
            ret[1] = left
        return ret